### DISCUSSION TITLE
Licence of pretrained models

### DISCUSSION THREAD

**Discussion by DanBmh** [NVIDIA-NeMo/NeMo#1681] [Q&A] [answered]
Hi, do the pretrained models from https://ngc.nvidia.com/catalog/models/nvidia:nemospeechmodels have the same Apache2 licence as this repository?

**Comment by okuchaiev** [comment, marked as answer]
1. NeMo software is released under [Apache 2.0 license](https://github.com/NVIDIA/NeMo/blob/main/LICENSE)
2. Pre-trained checkpoints which you can find on NVIDIA NGC are under [CC-BY-4](https://creativecommons.org/licenses/by/4.0/) unless another License is clearly specified for the model.

Going forward we intend to clearly specify the license within each model's "model card" so please pay attention to those.

**Reply by wasp77** [reply]
@okuchaiev I have read the NGC TERMS OF USE and it wasn't clear to me if the pre-trained checkpoints are allowed for commercial use. Any insight here would be much appreciated.

**Reply by okuchaiev** [reply]
@hardikmodi ^^

**Reply by opalrose510** [reply]
> Unless otherwise provided in the terms associated with Containers and Models, NVIDIA grants you a non-exclusive, non-transferable, non-sublicensable license, while you remain in compliance with the TOU, to modify and create derivative works of NVIDIA sample or example source code contained in Containers and Models. You hold all rights, title and interest in and to your modifications and derivative works of the NVIDIA sample and example source code software that you create as permitted hereunder (collective, “Derivatives”), subject to NVIDIA’s underlying intellectual property rights in and to the NVIDIA sample or example source code; provided, however that you grant NVIDIA and its affiliates an irrevocable, perpetual, nonexclusive, worldwide license to make, have made, use, have used, reproduce, license, distribute, sublicense, transfer and otherwise commercialize Derivatives without any obligation or compensation to you.


They are _allowed_ for commercial use, but NVIDIA may evidently at any time claim your model as theirs because it is a derivative of their model.

It seems like the vast majority of people would be surprised to discover this distinction, especially since the source allows open commercialization without restrictions. Is it possible to make this label (and the consequences) more clear, or consider relicensing the models, especially since they were trained on open source data?

**Reply by okuchaiev** [reply]
See the update to answer. Unless specified otherwise, NeMo models on NGC are under CC-BY-4.0 license. 
Going forward we plan on clearly specify a license with each individual model, so please pay attention to the model cards.

**Reply by gullabi** [reply]
@okuchaiev  Commenting from the future. It seems the NGC TOU has changed, I cannot find the above commented passage, neither any mention of an open license in the model cards. Are we to assume that the speech models in ngc catalog are no longer published under open licenses?

**Reply by GreatHalf** [reply]
It appears to me that all Nvidia models published on huggingface still have their cc-by-4.0 license.
NGC-exclusive models [like this one](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/riva/models/parakeet-ctc-riva-1-1b-unified-ml-cs-em-ea) however are harder to figure out.
